tests/testthat/test-community.R

```diff
@@ -0,0 +1,443 @@
+test_that("community detection functions work", {
+  withr::local_seed(42)
+
+  cluster_algos <- list(
+    "cluster_edge_betweenness", "cluster_fast_greedy",
+    "cluster_label_prop", "cluster_leading_eigen",
+    "cluster_louvain", "cluster_spinglass", "cluster_walktrap"
+  )
+  if (has_glpk()) cluster_algos <- c(cluster_algos, list("cluster_optimal"))
+
+  karate <- make_graph("Zachary")
+
+  for (algo in cluster_algos) {
+    cluster_algo <- get(algo)
+    comm <- cluster_algo(karate)
+
+    expect_equal(
+      modularity(comm),
+      modularity(karate, membership(comm))
+    )
+
+    karate_comunities <- communities(comm)
+    expect_true(all(!duplicated(unlist(karate_comunities))))
```

Comment:
why deduplicate if we use `all()`?

Reply:
changed to `any(duplicated(unlist(karate_comunities)))` as a more direct apporach. What this check does is to make sure that no vertex is in more than one cluster

Reply:
resolved by comment below :)